### PR TITLE
chore(pdata): add view-based OTLP JSON serialization

### PR DESCRIPTION
**Pull request by lquerel**
# Change Summary

Adds backend-agnostic OTLP JSON serialization for logs, metrics, and traces using pdata views. Supports owned protobuf, raw protobuf, and OTAP Arrow backends without materializing intermediate messages.

## How are these changes tested?

- Added 11 focused serialization and backend-parity tests.

## Are there any user-facing changes?

No. This adds an internal serialization API without changing component behavior or configuration.

### Changelog

- [ ] Added a .chloggen/*.yaml entry
- [x] This PR is a chore (indicated in title)
- [ ] This is a documentation-only PR.

### CODE REVIEW

File: rust/otap-dataflow/crates/pdata/src/views/otlp/bytes/decode.rs [hunk 1]
```diff
@@ -8,8 +8,54 @@ use std::marker::PhantomData;
...
 use std::rc::Rc;
 
+use crate::error::Error;
 use crate::proto::consts::wire_types;
 
+/// Validates the wire framing of one protobuf message without decoding nested messages.
+pub(super) fn validate_message_wire_format(buf: &[u8]) -> Result<(), Error> {
+    let mut pos = 0;
+    while pos < buf.len() {
+        let (tag, next) = read_varint(buf, pos).ok_or(Error::InvalidProtobufWireFormat)?;
```
Comment (lalitb) [inline rust/otap-dataflow/crates/pdata/src/views/otlp/bytes/decode.rs:18]: Small edge case: could we reject field keys larger than `u32::MAX` here? They are outside the valid protobuf key range, but `read_varint` can currently accept them as a u64. A small  negative test would be helpful too.

File: rust/otap-dataflow/crates/pdata/src/otlp/json/README.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+a semantic OTLP validator: callers remain responsible for constraints such as
+identifier widths, timestamp validity, and metric consistency.
+
+When a new OTLP field is added, expose it through the relevant view trait and
+all supported backends before adding it to the serializer. Backend-parity tests
+should cover the new field.
+
+## Related documentation
+
+- [OTLP JSON Protobuf specification](../../../../../../../proto/opentelemetry-proto/docs/specification.md)
```
Comment (lalitb) [inline rust/otap-dataflow/crates/pdata/src/otlp/json/README.md]: nit - this seems to be failing CI. Probably replace with canonical HTTPS link.

**Review by lalitb — APPROVED** [review]
LGTM - The shared encoding path across owned, raw, and Arrow views looks good. I left one small validation comment; otherwise this looks good to me.

File: rust/otap-dataflow/crates/pdata/src/otlp/json/metrics.rs [hunk 3]
```diff
@@ -0,0 +1,726 @@
...
+        }
+        let time = self.0.time_unix_nano();
+        if time != 0 {
+            map.serialize_entry("timeUnixNano", &ProtoU64(time))?;
+        }
+        if let Some(value) = self.0.value() {
+            serialize_value(&mut map, value)?;
+        }
+        if let Some(span_id) = self.0.span_id() {
+            map.serialize_entry("spanId", &hex::encode(span_id))?;
```
Comment (utpilla) [inline rust/otap-dataflow/crates/pdata/src/otlp/json/metrics.rs]: Each `hex::encode(...)` call allocates a fresh String just to hand serde a `&str`.  We could remove this allocation by encoding these IDs into small stack buffer and serialize that directly.

```rust
struct HexId<'a>(&'a [u8]);

impl Serialize for HexId<'_> {
    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
        const HEX: &[u8; 16] = b"0123456789abcdef";
        let mut buf = [0u8; 32]; // 16-byte trace_id -> 32 hex chars
        let n = self.0.len().min(16);
        for (i, &b) in self.0[..n].iter().enumerate() {
            buf[i * 2] = HEX[(b >> 4) as usize];
            buf[i * 2 + 1] = HEX[(b & 0x0f) as usize];
        }
        serializer.serialize_str(std::str::from_utf8(&buf[..n * 2]).map_err(S::Error::custom)?)
    }
}
```

**Review by utpilla — APPROVED** [review]

File: rust/otap-dataflow/crates/pdata/src/otlp/json/common.rs [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+            .ok_or_else(|| S::Error::custom("missing array AnyValue"))?;
+        let mut map = serializer.serialize_map(None)?;
+        if self
+            .0
+            .as_array()
+            .is_some_and(|mut values| values.next().is_some())
+        {
+            map.serialize_entry("values", &AnyValueIterJson::new(values))?;
+        }
+        map.end()
```
[the remark this answers is not in the document]
Comment (lquerel) [reply rust/otap-dataflow/crates/pdata/src/otlp/json/common.rs:285]: Updated both composite serializers to create the view iterator once, inspect it with Peekable::peek, and serialize that same iterator.

File: rust/otap-dataflow/crates/pdata/src/views/otlp/bytes/decode.rs (hunk 1 above)
Comment (lquerel) [reply rust/otap-dataflow/crates/pdata/src/views/otlp/bytes/decode.rs:18]: Added the u32::MAX key bound and a negative test covering the varint encoding of 2^32.

File: rust/otap-dataflow/crates/pdata/src/otlp/json/README.md (hunk 2 above)
Comment (lquerel) [reply rust/otap-dataflow/crates/pdata/src/otlp/json/README.md]: Replaced the submodule-relative link with the canonical HTTPS specification URL and validated the Markdown.

File: rust/otap-dataflow/crates/pdata/src/otlp/json/metrics.rs (hunk 3 above)
Comment (lquerel) [reply rust/otap-dataflow/crates/pdata/src/otlp/json/metrics.rs]: Done in https://github.com/open-telemetry/otel-arrow/pull/3764/commits/605006fbfc47b12fd1cabdcae11d57ca0c1bc89d
